### PR TITLE
[WEB-3823] E2E fixtures

### PR DESCRIPTION
**Pull request by tadam313**
_Disclaimer: Cypress does not support routing based on body params :(_

- moved visit to a custom cypress command (response wiretapped and replaced with local json)
- introduces eslint for typescript files (fix immediate errors)
- made possible to run E2E tests on local dev server

### CODE REVIEW

File: spec/integration/common.js [hunk 1]
```diff
@@ -98,19 +98,7 @@ export const should = (element, expectation) => {
 };
 
 Given("editor is open", () => {
-	cy.server()
-		.route("POST", "/1/indexes/steplib_steps/browse**")
-		.as("steplib-steps")
-		.route("POST", "/1/indexes/steplib_inputs/browse**")
-		.as("steplib-inputs");
-
 	cy.visit(`http://localhost:${PORT}/${version}/#!/workflows`);
```
Comment (tadam313) [inline spec/integration/common.js:101]: ported it to be a command

File: spec/integration/cypress/commands.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	cy.server();
+	cy.route("POST", "/1/indexes/steplib_inputs/**").as("steplib-inputs");
+
+	cy.fixture("steps.json").then(steps => {
+		cy.route(mockRequest("POST", "/1/indexes/steplib_steps/**", (xhr) => {
+			const { body } = xhr.request;
+			if (body.query === "") {
+				return steps;
+			}
+		}))
```
Comment (tadam313) [inline spec/integration/cypress/commands.js:23]: replacing response, when Cypress supports this use case we can migrate to that. (AFAIK it's in progress)

File: apiserver/routes.go [hunk 3]
```diff
@@ -48,14 +50,16 @@ func SetupRoutes(isServeFilesThroughMiddlemanServer bool) (*mux.Router, error) {
...
+		assetServer = httputil.NewSingleHostReverseProxy(u)
 	} else {
 		box := rice.MustFindBox("www")
-		http.Handle("/"+version.VERSION+"/", http.StripPrefix("/"+version.VERSION+"/", http.FileServer(box.HTTPBox())))
+		assetServer = http.StripPrefix("/"+version.VERSION+"/", http.FileServer(box.HTTPBox()))
 	}
 	//
 
 	http.Handle("/", r)
+	http.Handle("/"+version.VERSION+"/", assetServer)
```
Comment (tadam313) [inline apiserver/routes.go:62]: we could route versions number locally as well

File: webpack.config.js [hunk 4]
```diff
@@ -51,10 +51,13 @@ module.exports = {
 
 	devServer: {
 		contentBase: OUTPUT_FOLDER,
+		contentBasePublicPath: publicPath,
 		compress: true,
 		port: 4567
 	},
 
+	devtool: "inline-source-map",
```
Comment (tadam313) [inline webpack.config.js:59]: 🎉

File: source/javascripts/services/semver-service.ts [hunk 5]
```diff
@@ -101,7 +101,7 @@ class SemverService {
 
 		const stepVersions = Object.keys(stepCatalogue.steps[stepId]).sort(this.reverseSorter);
 
-		return stepVersions.find(stepVersion => this.isVersionCompatible(<string>version, stepVersion));
+		return stepVersions.find(stepVersion => this.isVersionCompatible(version as string, stepVersion));
```
Comment (simonmarton) [inline source/javascripts/services/semver-service.ts:104]:
```suggestion
		return stepVersions.find(stepVersion => this.isVersionCompatible(version!, stepVersion));
```

**Pull request merged**